### PR TITLE
Handle data type or empty string in module_utils

### PR DESCRIPTION
**Pull request by rjeffman**
Some parameters, in modules, have a specific data type, but allow the use of an empty string to clear the parameter.
    
By providing a method to retrieve the parameter with the correct data type, or optionally an empty string, allows for consistency of parameter handling between different modules.

Currently, `ipapwpolicy` and `ipaidoverrideuser` requires such method, and both were modified to use the module_utils implementation.

### CODE REVIEW

File: plugins/modules/ipapwpolicy.py [hunk 1]
```diff
@@ -294,20 +294,34 @@ def main():
...
+    maxrepeat = ansible_module.params_get_tipe(
+        "maxrepeat", int, allow_empty_string=True)
+    maxsequence = ansible_module.params_get_tipe(
+        "maxsequence", int, allow_empty_string=True)
+    dictcheck = ansible_module.params_get_tipe(
+        "dictcheck", bool, allow_empty_string=True)
+    usercheck = ansible_module.params_get_tipe(
+        "usercheck", bool, allow_empty_string=True)
+    gracelimit = ansible_module.params_get_tipe(
+        "gracelimit", int, allow_empty_string=True)
```
[the remark this answers is not in the document]
Comment (rjeffman) [reply plugins/modules/ipapwpolicy.py]: fixed.

File: plugins/module_utils/ansible_freeipa_module.py [hunk 2]
```diff
@@ -497,6 +497,49 @@ def module_params_get_lowercase(module, name, allow_empty_list_item=False):
     return value
 
 
+def module_params_get_type(
```
Comment (t-woerner) [inline plugins/module_utils/ansible_freeipa_module.py:500]: The `module_params_get_type` is misleading. Maybe something like `module_params_get_with_type` would be better.

File: plugins/module_utils/ansible_freeipa_module.py [hunk 3]
```diff
@@ -1072,6 +1115,28 @@ def params_get_lowercase(self, name, allow_empty_list_item=False):
         """
         return module_params_get_lowercase(self, name, allow_empty_list_item)
 
+    def params_get_type(
```
Comment (t-woerner) [inline plugins/module_utils/ansible_freeipa_module.py:1118]: Same as above.

**Review by varunmylaraiah — APPROVED** [review]
LGTM, Downstream tests are passed

**Comment by rjeffman** [comment]
@t-woerner what do you think about calling the method "params_get_with_type_cast"?

Not only this is more explicitly tells what the method is doing, but it also alows for new usage, such as `params_get_with_type_cast("aduser", ad_user_type)`, or `params_get_with_type_cast("service", list_of(service(ipa_realm))`.

I have the second example in a local "playground branch", and I'm getting fond of the solution for ensuring idempotence in the face of different value types.

(Note: I converted the PR to "draft" so that it is not merged before we are aligned on the method name.)

**Comment by rjeffman** [comment]
As discussed offline, I renamed the new method/function to `*_get_type_with_cast`, as it explicity says what it is doing, specially in the cases used by PR #1190 (which is based on these changes)

**Pull request merged**